lib/assert.js

```diff
@@ -284,12 +285,12 @@ function getErrMessage(message, fn) {
   const tmpLimit = Error.stackTraceLimit;
   // Make sure the limit is set to 1. Otherwise it could fail (<= 0) or it
   // does to much work.
-  Error.stackTraceLimit = 1;
+  if (!ObjectIsFrozen(Error)) Error.stackTraceLimit = 1;
```

Comment:
Do we need to perform this check multiple times in this function and is it possible to do it once during `require()` of this module instead of during each function call?

Reply:
@mscdex unfortunately this could change over the lifetime of the application if someone does `Object.freeze(Error)` in their own code :-( . Likely we could replace it with a helper fn that replaces itself with a noop once it sees `stackTraceLimit` is non-configurable.

Reply:
`ObjectIsFrozen` is not the correct check because it only returns `true` when the object is not extensible and all properties are non‑`writable` and non‑`configurable`.

---

The correct check is:
```js
function IsErrorStackTraceLimitWritable() {
	const desc = ObjectGetOwnPropertyDescriptor(Error, "stackTraceLimit");
	if (desc === undefined) {
		return ObjectIsExtensible(Error);
	}

	const { writable } = desc;
	if (writable === undefined) {
		return typeof desc.set === "function";
	}

	return writable;
}
```